### ISSUE TITLE
suggestion: add  rename function

### ISSUE THREAD

**Issue by zhangjg** [rethinkdb/rethinkdb#5858]
Now rehtinkdb have `dbCreate`, `dbDrop`, `tableCreate` and `tableDrop`, so we can create, delete the whole database or table, but there is no way to rename the  database and/or table!

So maybe we can add the `dbRename` , `tableRename` in the feature.

**Comment by zhangjg** [comment]
Sorry about this.
I find the `config`.

**Comment by danielmewes** [comment]
Yeah writing to the `db_config` or `table_config` system tables in the `rethinkdb` database is currently the way for doing this. It might not be the worst idea to add `rename` commands for databases and tables to ReQL. We already have an `indexRename` command for example.

I'm going to re-open this as a proposal for some future version.
We have some other proposals that suggest changing the `*Create`, `*Drop` interface for tables, which might or might not affect this. If we end up deciding on https://github.com/rethinkdb/rethinkdb/issues/3130 for example, the syntax for renaming should probably become `r.table(...).rename(new_name)`. We should hold off from discussing this for now I think.
